qiskit/dagcircuit/dagcircuit.py

```diff
@@ -702,14 +705,9 @@ def __eq__(self, other):
         # TODO this works but is a horrible way to do this
         slf = copy.deepcopy(self._multi_graph)
         oth = copy.deepcopy(other._multi_graph)
-
-        for node in slf.nodes:
-            slf.nodes[node]['node'] = node
-        for node in oth.nodes:
-            oth.nodes[node]['node'] = node
-
-        return nx.is_isomorphic(slf, oth,
-                                node_match=lambda x, y: DAGNode.semantic_eq(x['node'], y['node']))
+        return rx.is_isomorphic_node_match(slf, oth,
```

Comment:
Maybe the deep copies (and the comment :) ) can be removed here. They were needed because `nx.is_isomorphic` provides `node_match` with the node attributes and not the graph nodes, so we annotated each graph node with a reference to itself. I'm not sure if retworkx follows the same behavior.

Reply:
Heh, I agree the comment is probably not needed anymore. But, when I tested this without the deepcopies I got some weird hangs/performance issues running tests locally. I can try it again without them, since I added the deepcopy calls back prior to retworkx 0.2.0

Reply:
I just tested this locally without deep copies on the current version of the branch and it both slows down test execution (mostly because of `test_examples`) and causes a panic: `thread '<unnamed>' panicked at 'index out of bounds: the len is 4 but the index is 4'` in the isomorphism function. This is because of a bug in retworkx's isomorphism check function which I just copy and pasted form petgraph and changed the input type. It's failing to deal with comparing a `StableGraph` which has a monotonically increasing index (vs a regular `Graph` which will reuse an index from a removed node). Having the deepcopy works around this issue because a deepcopy resets the index counts when it reconstructs the graph (see: https://github.com/mtreinish/retworkx/blob/master/src/lib.rs#L259-L294 )

Anyway I've added an issue about this on retworkx: https://github.com/mtreinish/retworkx/issues/27 and I'll update the comment here to reflect why it's needed.